### PR TITLE
Retry for app user schema property read in create and update

### LINKED ISSUE #425: okta_app_user_schema inconsistent apply
An inconsistent apply is detected for the resource okta_app_user_schema when applied inside `count`. 

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.14.9

### Affected Resource(s)

- `okta_app_user_schema`

### Terraform Configuration Files

```hcl
variable "custom_attributes" {
  type = list(object({name = string, displayName = string, type = string, required = bool}))
 default = [ 
  {
      name : "firstName",
      type : "string",
      displayName : "First Name",
      required : true
    },
    {
      name : "lastName",
      type : "string",
      displayName : "Last Name",
      required : true
    }
 ]
}
resource "okta_app_user_schema" "custom_attributes" {
  count       =  2
  app_id      = "abcd"
  index       = var.custom_attributes[count.index].name
  title       = var.custom_attributes[count.index].displayName
  type        = var.custom_attributes[count.index].type
  description = var.custom_attributes[count.index].displayName
  required    = var.custom_attributes[count.index].required
  master      = "OKTA"
  scope       = "SELF"
  permissions = "READ_WRITE"
}
```

### Debug Output

N/A

### Panic Output

image

### Expected Behavior

Consistent applies

### Actual Behavior

Inconsistent apply detected

### Steps to Reproduce

1. `terraform apply`

### References

- #0000

### PR DESCRIPTION
**Pull request by ashwini-desai**
Fix #425

### CODE REVIEW

**Comment by ashwini-desai** [comment]
@bogdanprodan-okta Referred your [PR](https://github.com/okta/terraform-provider-okta/pull/390) to fix this.

File: go.mod [hunk 1]
```diff
@@ -7,10 +7,13 @@ require (
 	github.com/cenkalti/backoff/v4 v4.1.0
 	github.com/client9/misspell v0.3.4
 	github.com/crewjam/saml v0.4.5
+	github.com/google/go-cmp v0.5.4 // indirect
```
Comment (bogdanprodan-okta) [inline go.mod]: No need to include indirect packages in go mod

**Comment by bogdanprodan-okta** [comment]
Other than go.mod, everything looks good.

File: go.mod (hunk 1 above)
Comment (ashwini-desai) [reply go.mod]: @bogdanprodan-okta   Updated the go mod

**Comment by ashwini-desai** [comment]
@bogdanprodan-okta When could this be released? We are actually blocked on this.

**Review by bogdanprodan-okta — APPROVED** [review]

**Comment by bogdanprodan-okta** [comment]
> @bogdanprodan-okta When could this be released? We are actually blocked on this.

I'm planning Github Release by the end of this month. As for the terraform registry - no idea. Still no updates from their side.

**Pull request merged**